oap-server/server-query-plugin/prometheus-query-plugin/src/main/java/org/apache/skywalking/oap/query/prometheus/rt/PromOpUtils.java

```diff
@@ -0,0 +1,255 @@
+                    "The metric info result left in conformity with right.");
+            }
+            if (dataLeft.getValues().size() != dataRight.getValues().size()) {
+                throw new IllegalExpressionException(
+                    "The metric value range left in conformity with right.");
+            }
+            MetricRangeData newData = new MetricRangeData();
+            result.getMetricDataList().add(newData);
+            newData.setMetric(dataLeft.getMetric());
+            List<TimeValuePair> newValues = new ArrayList<>();
+            newData.setValues(newValues);
+            for (int j = 0; j < dataLeft.getValues().size(); j++) {
+                double lv = Double.parseDouble(dataLeft.getValues().get(j).getValue());
+                double rv = Double.parseDouble(dataRight.getValues().get(j).getValue());
+                newValues.add(new TimeValuePair(
+                    dataLeft.getValues().get(j).getTime(),
+                    formatDoubleValue(scalarBinaryOp(lv, rv, opType))
+                ));
+            }
+        }
+        return result;
+    }
+
+    static double scalarBinaryOp(double leftValue, double rightValue, int opType) {
+        double calculatedResult = 0;
+        switch (opType) {
+            case PromQLParser.ADD:
+                calculatedResult = leftValue + rightValue;
+                break;
+            case PromQLParser.SUB:
+                calculatedResult = leftValue - rightValue;
+                break;
+            case PromQLParser.MUL:
+                calculatedResult = leftValue * rightValue;
+                break;
+            case PromQLParser.DIV:
+                calculatedResult = leftValue / rightValue;
+                break;
+            case PromQLParser.MOD:
+                calculatedResult = leftValue % rightValue;
+                break;
+        }
+        return calculatedResult;
+    }
+
+    static int scalarCompareOp(double leftValue, double rightValue, int opType) {
+        int comparedResult = 0;
+        switch (opType) {
+            case PromQLParser.DEQ:
+                comparedResult = boolToInt(leftValue == rightValue);
+                break;
+            case PromQLParser.NEQ:
+                comparedResult = boolToInt(leftValue != rightValue);
+                break;
+            case PromQLParser.GT:
+                comparedResult = boolToInt(leftValue > rightValue);
+                break;
+            case PromQLParser.LT:
+                comparedResult = boolToInt(leftValue < rightValue);
+                break;
+            case PromQLParser.GTE:
+                comparedResult = boolToInt(leftValue >= rightValue);
+                break;
+            case PromQLParser.LTE:
+                comparedResult = boolToInt(leftValue <= rightValue);
+                break;
+        }
+        return comparedResult;
+    }
+
+    private static int boolToInt(Boolean v) {
+        return v ? 1 : 0;
+    }
```

Comment:
Either change the type of the argument to `boolean` or add null check for `v`, otherwise when `v == null` NPE occurs here